src/test/java/org/jenkinsci/plugins/durabletask/BourneShellScriptTest.java

```diff
@@ -236,4 +239,141 @@ private void runOnDocker(DumbSlave s) throws Exception {
         runOnDocker(new DumbSlave("docker", "/home/jenkins/agent", new SimpleCommandLauncher("docker run -i --rm --name agent --init jenkinsci/slave:3.7-1 java -jar /usr/share/jenkins/slave.jar")));
     }
 
+    @Issue("JENKINS-31096")
+    @Test public void encoding() throws Exception {
```

Comment:
This test is unreadable IMHO. Would it be possible to split it to several tests or at least explicitly indicate test stages? Assert logic could be also refactored to a separate method

Reply:
Yes, it could be refactored to use a helper assertion method with parameters.

Reply:
(resolved)